src/crewai/agents/agent_builder/base_agent.py

```diff
@@ -181,7 +181,7 @@ def key(self):
             self._original_goal or self.goal,
             self._original_backstory or self.backstory,
         ]
-        return md5("|".join(source).encode(), usedforsecurity=False).hexdigest()
+        return sha256("|".join(source).encode()).hexdigest()
```

Comment:
I've used md5 here because it's not supposed to be protected data, but simply a hash of strings. The md5 algorithm is supposed to be fast as safe for this use case. Can we safely replace it with sha256 and achieve the same performance?